samples/unit-tests/fullscreen/toggle/demo.js

```diff
@@ -29,3 +29,34 @@ QUnit.test('Fullscreen module.', function (assert) {
 
     assert.ok(true, 'Chart displayed in fullscreen mode without any errors.');
 });
+
+function sleep(milliseconds) {
+    return new Promise(resolve => setTimeout(resolve, milliseconds));
+}
+
+QUnit.test('#20548, chart resizing after fullscreen.', async function (assert) {
+    const done = assert.async(),
+        chart = Highcharts.chart('container', {
+            series: [
+                {
+                    data: [5, 3, 4, 2, 4, 3]
+                }
+            ]
+        });
+    const oldChartWidth = chart.chartWidth;
+    const oldXAxisLength = chart.axes[0].len;
+
+    chart.fullscreen.open();
+    chart.fullscreen.close();
+
+    // Simulates a container resize, e.g. like a window resize.
+    chart.container.parentElement.style.width = '300px';
+    // Let the browser update styles.
+    await sleep(250);
```

Comment:
This is a quite long waiting time. I understand it needs to be more than 100ms for the setSize in reflow to finish as it runs after a fixed (100ms) interval, but maybe this could be more stable if the below assertion were running after chart's `endResize` event that is fired when the setSize finishes? Having a fallback to some fixed timeout in case `reflow` doesn't call `setSize` at all would be good.